### PR TITLE
View component tools support

### PR DESCRIPTION
**Pull request by body-clock**
Addresses #3102 

ViewComponent 3.0.0 switched the Slots API and how they're called. This PR addresses some code that had not been updated to reflect those changes. It also aims to test that the Tools components render their modals properly to prevent similar issues in the future.

Just one note - because the citation tool relies on methods not present in core Blacklight, the tool's functionality cannot test that the modal content renders correctly. I've chosen to just test that the header renders correctly. Through my investigation, I've also noticed that the [Blacklight Demo](https://demo.projectblacklight.org/catalog/2009002583) doesn't have a working citation tool - something to think about.

### CODE REVIEW

**Comment by jcoyne** [comment]
Will this still work with view_component 2?  The gemspec indicates that 2 is an acceptable version.

File: spec/features/tools_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Tools' do
+  before { visit solr_document_path('2007020969') }
+
+  it 'displays Email modal with form' do
+    click_link 'Email'
```
Comment (jcoyne) [inline spec/features/tools_spec.rb:7]: Isn't this a duplicate case as the `spec/features/modal_spec.rb`?

**Review by jcoyne — CHANGES_REQUESTED** [review]

**Comment by body-clock** [comment]
> Will this still work with view_component 2? The gemspec indicates that 2 is an acceptable version.

I'm having a hard time figuring this out. The ViewComponent changelog for 3.0 lists this as a breaking change, but it seems this convention has been adopted throughout the rest of the Blacklight gem. For example `app/views/catalog/facet.html.erb` uses `component.with_prefix`, `component.with_title`, and `component.with_footer`, following the conventions of VC 3.

**Comment by jrochkind** [comment]
> Will this still work with view_component 2? The gemspec indicates that 2 is an acceptable version.

@jcoyne , the gemspec requires at least view_component 2.66, not actually any 2. 

The "newer" `with_` API is available from 2.54.0 on. https://viewcomponent.org/CHANGELOG.html#2540

So should be good!

**Comment by jcoyne** [comment]
Thank you for taking the lead on this @body-clock.  I think I've incorporated all this into #3120 and merged into main.

**Pull request closed without merging**